File: src/game/phases/phases.lua
local Phase = {}
Phase.__index = Phase

function Phase:new(name, start_actions, end_actions, allowed_action_types, allowed_effect_speeds)
    local phase = setmetatable({}, Phase)

    phase.name = name
    phase.start_actions = start_actions or {}
    phase.end_actions = end_actions or {}
    phase.allowed_actions = allowed_action_types or {}
    phase.allowed_effect_speeds = allowed_effect_speeds or {}

    return phase
end

return Phase

File: src/server/game/config/phases.go
package config

import "sync"

type Phase struct {
	Name                string   `json:"name"`
	StartActions        []string `json:"start_actions"`
	EndActions          []string `json:"end_actions"`
	AllowedActionTypes  []string `json:"allowed_action_types"`
	AllowedEffectSpeeds []string `json:"allowed_effect_speeds"`
}

type PhasesConfig struct {
	Phases []Phase `json:"phases"`
}

var (
	phasesConfig    sync.Once
	PhasesConfigVar PhasesConfig
)

// GetPhasesConfig allows to get the game configuration linked to phases globally.
func GetPhasesConfig() PhasesConfig {
	return PhasesConfigVar
}
